### PR TITLE
Add energy group structure for ANL21 and ANL94 used in photon transport

### PR DESCRIPTION
**Pull request by keckler**
## Description

This just adds another typical energy group structure to our physics.neutronics.energyGroups module. The added structure is common for photon transport.

---

## Checklist

- [x] This PR has only one purpose or idea.
- [ ] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by jakehader** [comment]
Thanks @keckler - can you add the ANL94 group for photons too?

**Comment by keckler** [comment]
@jakehader Sure, done!

File: armi/physics/neutronics/energyGroups.py [hunk 1]
```diff
@@ -774,3 +774,132 @@ def _create_anl_energies_with_group_energies(group_energy_bounds):
         4.140e-01,
     ]
 )
+
+"""
+Taken from Table 5.1 of "GAMSOR: Gamma Souce Preparation and DIF3D Flux Solution",
+ANL/NE-16/50 Rev 2.0, M.A. Smith, C.H. Lee, R.N. Hill, Aug 30 2022.
+"""
+GROUP_STRUCTURE["ANL21PHOTON"] = [
```
Comment (jakehader) [inline armi/physics/neutronics/energyGroups.py:782]: This is typically denoted as `"ANL21G"` recommend updating.

File: armi/physics/neutronics/energyGroups.py [hunk 2]
```diff
@@ -774,3 +774,132 @@ def _create_anl_energies_with_group_energies(group_energy_bounds):
...
+    4.5e4,
+    3.0e4,
+    2.0e4,
+]
+
+"""
+Taken from Table 5.2 of "GAMSOR: Gamma Souce Preparation and DIF3D Flux Solution",
+ANL/NE-16/50 Rev 2.0, M.A. Smith, C.H. Lee, R.N. Hill, Aug 30 2022.
+"""
+GROUP_STRUCTURE["ANL94PHOTON"] = [
```
Comment (jakehader) [inline armi/physics/neutronics/energyGroups.py:810]: This is typically denoted as `"ANL94G"` recommend updating.

**Review by jakehader — CHANGES_REQUESTED** [review]
Numbers look correct! Minor comment.

**Comment by keckler** [comment]
@jakehader Names are fixed.

**Review by jakehader — APPROVED** [review]

**Pull request merged**